monai/losses/focal_loss.py

```diff
@@ -234,9 +234,8 @@ def sigmoid_focal_loss(
     """
     # computing binary cross entropy with logits
     # equivalent to F.binary_cross_entropy_with_logits(input, target, reduction='none')
-    # see also https://github.com/pytorch/pytorch/blob/v1.9.0/aten/src/ATen/native/Loss.cpp#L231
-    max_val = (-input).clamp(min=0)
-    loss: torch.Tensor = input - input * target + max_val + ((-max_val).exp() + (-input - max_val).exp()).log()
+    # see also https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/Loss.cpp#L363
+    loss: torch.Tensor = input - input * target - F.logsigmoid(input)
```

Comment:
Is this change just to speed up our code? It mirrors the PR in Pytorch you mentioned in the issue.

Reply:
To align the latest PyTorch, and when the latest PyTorch is used, the max_error will becomes zero which means our `FocalLoss` will get the same result with `BCEWithLogitsLoss` when gamma is zero.